src/sshagent/SSHAgent.cpp

```diff
@@ -53,9 +55,14 @@ SSHAgent* SSHAgent::instance()
     return m_instance;
 }
 
-void SSHAgent::init(QObject* parent)
+void SSHAgent::init(MessageWidget* messageWidget)
 {
-    m_instance = new SSHAgent(parent);
+    m_instance = new SSHAgent(messageWidget);
```

Comment:
It don't like parenting the SSH agent to a single GUI widget. You should continue using the MainWindow or, even better, the application itself. Either pass in the MessageWidget as a second mandatory parameter or, even better, remove tight coupling altogether and only emit signals which you can connect to the MessageWidget from the outside.